scripts/src/javatests/org/oppia/android/scripts/maven/GenerateMavenDependenciesListTest.kt

```diff
@@ -0,0 +1,423 @@
+import org.oppia.android.scripts.proto.MavenDependency
+import org.oppia.android.scripts.proto.MavenDependencyList
+import org.oppia.android.scripts.proto.ScrapableLink
+import org.oppia.android.scripts.testing.TestBazelWorkspace
+import org.oppia.android.testing.assertThrows
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.PrintStream
+
+/** Tests for [GenerateMavenDependenciesList]. */
+class GenerateMavenDependenciesListTest {
+
+  private val DATA_BINDING_POM = "https://maven.google.com/androidx/databinding/" +
+    "databinding-adapters/3.4.2/databinding-adapters-3.4.2.pom"
+  private val PROTO_LITE_POM = "https://repo1.maven.org/maven2/com/google/protobuf/" +
+    "protobuf-lite/3.0.0/protobuf-lite-3.0.0.pom"
+  private val IO_FABRIC_POM = "https://maven.google.com/io/fabric/sdk/android/" +
+    "fabric/1.4.7/fabric-1.4.7.pom"
+  private val GLIDE_ANNOTATIONS_POM = "https://repo1.maven.org/maven2/com/github/" +
+    "bumptech/glide/annotations/4.11.0/annotations-4.11.0.pom"
+  private val FIREBASE_ANALYTICS_POM = "https://maven.google.com/com/google/firebase/" +
+    "firebase-analytics/17.5.0/firebase-analytics-17.5.0.pom"
+
+  private val THIRD_PARTY_PREFIX = "//third_pary:"
+  private val DATA_BINDING_COORD = "androidx.databinding:databinding-adapters:3.4.2"
+  private val PROTO_LITE_COORD = "com.google.protobuf:protobuf-lite:3.0.0"
+  private val GLIDE_ANNOTATIONS_COORD = "com.github.bumptech.glide:annotations:4.11.0"
+  private val FIREBASE_ANALYTICS_COORD = "com.google.firebase:firebase-analytics:17.5.0"
+  private val IO_FABRIC_COORD = "io.fabric.sdk.android:fabric:1.4.7"
+
+  private val DATA_BINDING_COORD_VERSION = "3.4.2"
+  private val PROTO_LITE_COORD_VERSION = "3.0.0"
+  private val GLIDE_ANNOTATIONS_COORD_VERSION = "4.11.0"
+  private val FIREBASE_ANALYTICS_COORD_VERSION = "17.5.0"
+  private val IO_FABRIC_COORD_VERSION = "1.4.7"
+
+  private val LICENSE_DETAILS_INCOMPLETE_FAILURE = "Licenses details are not completed"
+  private val UNAVAILABLE_OR_INVALID_LICENSE_LINKS_FAILURE =
+    "License links are invalid or not available for some dependencies"
+  private val SCRIPT_PASSED_MESSAGE =
+    "Script executed succesfully: maven_dependencies.textproto updated successfully."
+
+  private val outContent: ByteArrayOutputStream = ByteArrayOutputStream()
+  private val originalOut: PrintStream = System.out
+
+  private val mockLicenseFetcher by lazy { initializeLicenseFetcher() }
+  private lateinit var testBazelWorkspace: TestBazelWorkspace
+
+  @Rule
+  @JvmField
+  var tempFolder = TemporaryFolder()
+
+  @Before
+  fun setUp() {
+    tempFolder.newFolder("scripts", "assets")
+    tempFolder.newFolder("third_party")
+    testBazelWorkspace = TestBazelWorkspace(tempFolder)
+    System.setOut(PrintStream(outContent))
+  }
+
+  @After
+  fun restoreStreams() {
+    System.setOut(originalOut)
+  }
+
+  @Test
+  fun testLicenseLinkNotVerified_scriptFailsWithException() {
```

Comment:
Is it that they aren't verified or that the files are empty? The test name should be really clear on the specific conditions that lead to this outcome.

Ditto for other test names.

Reply:
Actually, both, if the files are empty that means that the script has to be run for the very first time that also implies that the license details wouldn't be verified in this case.